### PR TITLE
Update libQtOlm

### PR DESCRIPTION
**Pull request by encombhat**
Changes in libQtOlm:
Remove FindOlm.cmake.
Use CMake find_package in config mode. It is the preferred method for looking for olm.

Note that due to these changes, Appveyor CI will not work. I will try to fix that.

### CODE REVIEW

**Comment by encombhat** [comment]
I'm not sure why the CI builds still failed.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
 cmake_minimum_required(VERSION 3.10)
+set(CMAKE_CXX_STANDARD 17)
```
Comment (KitsuneRal) [inline CMakeLists.txt]: What if I switch to C++20 and QtOlm happens to use things dropped from C++20? Also, I'm going to add QtKeychain to the list of the library dependencies; this change will mean that if QtKeychain is not compliant with whichever  CXX_STANDARD I use I'll have to patch it or wait for the upstream.

In other words, I'd prefer to not impose C++ level requirements on other libraries - rather, I expect the library to expose their C++ standard through respective PROPERTIES.

File: .appveyor.yml [hunk 2]
```diff
@@ -18,15 +18,18 @@ init:
...
+- cmake --build build
+- cmake --build build --target install
+- cd ..
+- cmake -G "NMake Makefiles JOM" -H. -Bbuild -DBUILD_SHARED_LIBS=OFF -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_INSTALL_PREFIX="%DEPLOY_DIR%" -DOlm_DIR="olm/install/lib/cmake/Olm"
 
 build_script:
 - cmake --build build
 # qmake uses olm just built by CMake - it can't build olm on its own.
-- qmake "INCLUDEPATH += 3rdparty/libQtOlm/olm/include" "LIBS += -Lbuild" && jom
+# - qmake "INCLUDEPATH += 3rdparty/libQtOlm/olm/include" "LIBS += -Lbuild" && jom
```
Comment (KitsuneRal) [inline .appveyor.yml]: ?

Comment (encombhat) [reply .appveyor.yml]: Oh sorry, QMake build is working, but I forgot to reenable it.

File: CMakeLists.txt (hunk 1 above)
Comment (encombhat) [reply CMakeLists.txt]: I'm not sure... The CI failed if CMAKE_CXX_STANDARD is not specified.

Comment (encombhat) [reply CMakeLists.txt]: Am I missing something?

Comment (KitsuneRal) [reply CMakeLists.txt]: That was broken even before your PR - qmc-example doesn't have CXX_STANDARD set, and the library's property is not exposed.

Just found that the right way of specifying the C++ standard seems to be via `target_compile_features()`: https://cmake.org/cmake/help/v3.10/manual/cmake-compile-features.7.html#requiring-language-standards.

Comment (encombhat) [reply CMakeLists.txt]: Ah, ok

**Pull request merged**